unittests/compressed_merkle_tests.cpp

```diff
@@ -0,0 +1,210 @@
+#include <eosio/amqp_compressed_proof_plugin/amqp_compressed_proof_plugin.hpp>
+#include <eosio/testing/tester.hpp>
+
+#include <boost/test/data/test_case.hpp>
+
+#include "compressed_proof_validator.hpp"
+
+using namespace eosio;
+using namespace eosio::chain;
+using namespace eosio::testing;
+using namespace fc;
+
+static const char fail_everything_wast[] = R"=====(
+(module
+ (export "apply" (func $apply))
+ (func $apply (param $0 i64) (param $1 i64) (param $2 i64)
+  (unreachable)
+ )
+)
+)=====";
+
+static const char test_account_abi[] = R"=====(
+{
+   "version": "eosio::abi/1.0",
+   "types": [],
+   "structs": [{"name":"dothedew", "base": "", "fields": [{"name":"nonce", "type":"uint32"}]}],
+   "actions": [{"name":"dothedew", "type": "dothedew", "ricardian_contract":""}],
+   "tables": [],
+   "ricardian_clauses": [],
+   "variants": [],
+   "action_results": []
+}
+)=====";
+
+static auto wire_signals_to_generator(compressed_proof_generator& generator, controller& ctrl) {
+   std::vector<boost::signals2::scoped_connection> conns;
+   conns.emplace_back(ctrl.irreversible_block.connect([&](const block_state_ptr& bsp) {
+      generator.on_irreversible_block(bsp, ctrl);
+   }));
+   conns.emplace_back(ctrl.applied_transaction.connect([&](std::tuple<const transaction_trace_ptr&, const packed_transaction_ptr&> t) {
+      generator.on_applied_transaction(std::get<0>(t));
+   }));
+   conns.emplace_back(ctrl.accepted_block.connect([&](const block_state_ptr& p) {
+      generator.on_accepted_block(p);
+   }));
+   return conns;
+}
+
+BOOST_AUTO_TEST_SUITE(test_compressed_proof)
```

Comment:
There should probably be at least one test case where the action schedule order is different from the execution order.

Reply:
Probably something with deferred too since they're special